eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -239,38 +211,41 @@ private void getOrCreateClusteredVariantAccessions(List<? extends SubmittedVaria
-                    assembly, allExistingHashesInDB, mergeCandidateSVOE);
+            boolean isExistingRSOrMergeCandidate = checkIfExistingRSOrCandidateForMerge(remappedSubmittedVariantEntity,
+                                                                                        clusteredVariantEntity,
+                                                                                        assembly, allExistingHashesInDB,
+                                                                                        mergeCandidateSVOE);
 
-            updateAllExistingHashesGroupByRS(allExistingHashesGroupByRS, assembly, clusteredVariantEntity.getAccession());
+            updateAllExistingHashesGroupByRS(allExistingHashesGroupByRS, assembly,
+                                             clusteredVariantEntity.getAccession());
 
-            boolean isSplitCandidate = checkIfCandidateForRSSplit(submittedVariantEntity, clusteredVariantEntity,
-                    assembly, allExistingHashesGroupByRS, rsSplitCandidateSVOE);
+            checkIfCandidateForRSSplit(remappedSubmittedVariantEntity, clusteredVariantEntity, assembly,
+                                       allExistingHashesGroupByRS, rsSplitCandidateSVOE);
 
-            if (!isMergeCandidate) {
-                if(clusteredVariantEntity.getAccession() >= accessioningMonotonicInitRs){
+            if (!isExistingRSOrMergeCandidate) {
+                if (clusteredVariantEntity.getAccession() >= accessioningMonotonicInitRs) {
                     clusteredVariantEntities.add(clusteredVariantEntity);
-                }else{
+                } else {
                     dbsnpClusteredVariantEntities.add(clusteredVariantEntity);
                 }
                 allExistingHashesGroupByRS.get(clusteredVariantEntity.getAccession())
-                        .add(clusteredVariantEntity.getHashedMessage());
-                allExistingHashesInDB.put(clusteredVariantEntity.getHashedMessage(), clusteredVariantEntity.getAccession());
+                                          .add(clusteredVariantEntity.getHashedMessage());
+                allExistingHashesInDB.put(clusteredVariantEntity.getHashedMessage(),
+                                          clusteredVariantEntity.getAccession());
             }
         }
 
         insertAllEntriesInDB(clusteredVariantEntities, dbsnpClusteredVariantEntities, mergeCandidateSVOE, rsSplitCandidateSVOE);
-
-        return new HashSet<>(clusteredRemappedSubmittedVariants);
     }
 
     private List<SubmittedVariantEntity> getClusteredAndRemappedVariants(List<? extends SubmittedVariantEntity> submittedVariants) {
         return submittedVariants.stream()
                 .filter(sve -> Objects.nonNull(sve.getClusteredVariantAccession()))
                 .filter(sve -> !StringUtil.isBlank(sve.getRemappedFrom()))
+                .filter(sve -> Objects.isNull(sve.getMapWeight()))
```

Comment:
Discard multimap variants from being clustered since they will be dealt with in EVA-2205.